core/src/trace/component.rs

```diff
@@ -0,0 +1,59 @@
+//! Abscissa tracing component
+
+// TODO(tarcieri): logfile support?
+
+use tracing_log::LogTracer;
+use tracing_subscriber::{fmt::Formatter, reload::Handle, EnvFilter, FmtSubscriber};
+
+use super::config::Config;
+use crate::{Component, FrameworkError, FrameworkErrorKind};
+
+/// Abscissa component for initializing the `tracing` subsystem
+#[derive(Component, Debug)]
+#[component(core)]
+pub struct Tracing {
+    filter_handle: Handle<EnvFilter, Formatter>,
+}
+
+impl Tracing {
+    /// Create a new [`Tracing`] component from the given [`Config`].
+    pub fn new(config: Config) -> Result<Self, FrameworkError> {
+        // Configure log/tracing interoperability by setting a `LogTracer` as
+        // the global logger for the log crate, which converts all log events
+        // into tracing events.
+        LogTracer::init().map_err(|e| FrameworkErrorKind::ComponentError.context(e))?;
+
+        // Construct a tracing subscriber with the supplied filter and enable reloading.
+        let builder = FmtSubscriber::builder()
+            // XXX this should use ColorChoice::should_ansi, but where
+            // do we get the color choice from?
+            .with_ansi(true)
+            .with_env_filter(config.filter)
```

Comment:
Dunno how @tarcieri feels about this, but I think it could be good to also honor RUST_LOG and allow it to override the CLI configs? I notice the CLI configs only allow enabling `info` and above or `debug` and above, while the RUST_LOG filtering allows much more granularity. I suppose if you also expose a dynamic filter endpoint, that would also allow more granularity.

Reply:
Yes please 👍 

Failure to do all of those things in the past was effectively just missing features.